agent/consul/intention_endpoint.go

```diff
@@ -837,15 +795,18 @@ func (s *Intention) Check(
 	// NOTE(mitchellh): This is the same behavior as the agent authorize
 	// endpoint. If this behavior is incorrect, we should also change it there
 	// which is much more important.
-	authz, err = s.srv.ResolveToken("")
-	if err != nil {
-		return err
+	defaultDecision := acl.Allow
+	if authz != nil {
```

Comment:
I wonder why we needed the anonymous  resolve before, if the `IntentionDefaultAllow` function can return the default acl policy